src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[fromFormat.length - 1];
+  const separatorTo = toFormat[toFormat.length - 1];
```

Comment:
The current logic assumes that the last element of `fromFormat` and `toFormat` is the separator. This might not be correct if the format arrays do not end with the separator. Consider explicitly specifying separators or using a different method to determine them.